File: game/game_objects/sticker.py
# Defines the data structure for Stickers.
from dataclasses import dataclass

@dataclass
class Sticker:
    """Represents a single sticker that belongs to an album."""
    id: str
    name: str
    album: str
    rarity: int  # e.g., 1 to 5 stars

    def __str__(self):
        return f"'{self.name}' Sticker (Album: {self.album}, Rarity: {self.rarity}*)"

File: unity_project/Assets/Scripts/Data/Sticker.cs
using System;

/// <summary>
/// Represents a single sticker that belongs to an album.
/// This class is a C# representation of the Python Sticker class.
/// In the current Python prototype, this class is defined but not used in any JSON data.
/// It's included here for completeness based on the Python source.
/// </summary>
[Serializable]
public class Sticker
{
    public string id;
    public string name;
    public string album;
    public int rarity; // e.g., 1 to 5 stars

    public override string ToString()
    {
        return $"'{name}' Sticker (Album: {album}, Rarity: {rarity}*)";
    }
}
